site/js/ChannelsActor.js

```diff
@@ -0,0 +1,106 @@
+import Actor from './Actor.js'
+import { get, post } from './api.js'
+
+export default class ChannelsActor extends Actor {
+  init() {
+    this.on('update channel list', channels => {
+      // Join the first channel if we're not already in one
+      if (this.activeChannelID === null) {
+        const defaultChannel = channels[0]
+
+        if (defaultChannel)
+          this.viewChannel(defaultChannel.id)
+      }
+
+      // Update UI
+      this.populateSidebarList(channels)
+    })
+
+    this.socket.on('created new channel', () => this.loadChannels())
+
+    this.on('update active channel', channel => {
+      document.title = '#' + channel.name
+      this.populateSidebarList(this.channels)
+    })
+
+    document.getElementById('create-channel').addEventListener('click', async evt => {
+      evt.preventDefault()
+
+      const channelName = prompt('New channel name?')
+      if (!channelName) return
+
+      const res = await post('create-channel', {
+        name: channelName,
+        sessionID: this.actors.session.sessionID,
+      })
+
+      if (res.error) {
+        alert('Error: ' + res.error)
```

Comment:
May want to say "error creating channel" rather than just "error" (e.g. if somebody has really, *really* low speed internet, and performs other operations while `await post` is going).

Also, I try to add more human-readable messages - `if (res.error === 'blah') { alert('Blah, blah blah blah. Try this. Blah.') }` - but this is okay for sure.